src/context.rs

```diff
@@ -92,29 +92,91 @@ pub fn get_info() -> HashMap<&'static str, String> {
     res
 }
 
-impl Context {
-    pub fn new(cb: Box<ContextCallback>, os_name: String, dbfile: PathBuf) -> Result<Context> {
-        let mut blob_fname = OsString::new();
-        blob_fname.push(dbfile.file_name().unwrap_or_default());
+/// Builder for [Context].
+///
+/// # Example
+///
+/// ```
+/// let dir = tempfile::tempdir().unwrap();
+/// let dbfile = dir.path().join("my-context.db");
+/// let ctx = ContextBuilder
+///     .new(Box::new(|_, _| 0), "AppName", dbfile)
+///     .blobdir(dir.path().join("my-context-blobs"))
+///     .blobdir(dir.path().join("my-context-logs"))
+///     .create();
+/// assert_eq!(ctx.get_blobdir(), dir.path().join("my-context-blobs"));
+/// ```
+#[derive(DebugStub)]
+pub struct ContextBuilder {
```

Comment:
https://crates.io/crates/derive_builder is a nice way to get rid of boilerplate and have nicer `Builder`s

Reply:
Building the defaults as we do, while I think not impossible, would not be clearer or easier I think.  So I left this.